device/isis/segmentrouting.yaml

```diff
@@ -0,0 +1,193 @@
+        Reference: https://datatracker.ietf.org/doc/html/rfc8667
+      type: object
+      properties:
+        router_capability:
+          description: |-
+            Optional IS-IS TLV named CAPABILITY, formed of multiple sub-TLVs, which allows a router to announce its
+            capabilities within an IS-IS level or the entire routing domain.        
+          $ref: '#/components/schemas/Isis.RouterCapability'
+          x-field-uid: 1
+      
+    Isis.RouterCapability:
+      description: >-
+        Container for the configuration of IS-IS Router CAPABILITY TLV.
+        https://datatracker.ietf.org/doc/html/rfc7981#section-2
+      type: object
+      properties:
+        choice:
+          description: |-
+            The Router Capability ID SHOULD be identical to the value advertised in the Traffic Engineering Router ID TLV [RFC5305].
+            If no Traffic Engineering Router ID is assigned, the Router ID SHOULD be identical to an IP Interface Address [RFC1195] 
+            advertised by the originating IS.
+            If the originating node does not support IPv4, then the reserved value 0.0.0.0 MUST be used in the Router ID field, 
+            and the IPv6 TE Router ID sub-TLV [RFC5316] MUST be present in the TLV.
+            - ipv4_te_router_id: IPv4 Traffic Engineering(TE) router id (defined in Isis.Basic class) to be used as Router Capability ID.
+            - interface_ip: When underlined IPv4 interface address of the ISIS interface to be assigned as Router Capability ID.
+            - custom: When, Router Capability ID needs to be configured different from above two.
+          type: string
+          default: ipv4_te_router_id
+          x-field-uid: 1
+          x-enum:
+            ipv4_te_router_id:
+              x-field-uid: 1
+            interface_ip:
+              x-field-uid: 2
+            custom_router_cap_id:
+              x-field-uid: 3
+        custom_router_cap_id:
+          description: >-
+            Router CapabilityID in IPv4 address format.
+          type: string
+          format: ipv4
+          x-field-uid: 2
+        flood:
+          description: |-
+            S bit (0x01): If the S bit is set(1), the IS-IS Router CAPABILITY TLV
+            MUST be flooded across the entire routing domain.  If the S bit is
+            not set(0), the TLV MUST NOT be leaked between levels.  This bit MUST
+            NOT be altered during the TLV leaking.
+          type: boolean
+          default: false
+          x-field-uid: 3
+        down:
+          description: |-
+            D bit (0x02): When the IS-IS Router CAPABILITY TLV is leaked from
+            Level 2 (L2) to Level 1 (L1), the D bit MUST be set.  Otherwise, this
+            bit MUST be clear.  IS-IS Router CAPABILITY TLVs with the D bit set
+            MUST NOT be leaked from Level 1 to Level 2.  This is to prevent TLV looping.
+          type: boolean
+          default: false
```

Comment:
Is false a good default ? Based on description, seems to allow possibility of TLV looping.